### PR TITLE
add solr for AI-telemetry

### PR DESCRIPTION
**Pull request by RH-csaggin**
- adding the basic components for 3 replicas solr with 3 shard
- this has a zookeeper svc dependency to work with
- created exteralsecret

### CODE REVIEW

**Review by tssala23 — APPROVED** [review]

File: ai-telemetry/base/solr/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+kind: Secret
+metadata:
+  name: solr
+  labels:
+    app.kubernetes.io/instance: solr
+    app.kubernetes.io/name: solr
+    app.kubernetes.io/component: solr
+type: Opaque
+data:
+  solr-password: "ZnVjOVpWNHl3Nw=="
```
Comment (computate) [inline ai-telemetry/base/solr/secret.yaml]: We should remove this secret, change the password, and make it an ExternalSecret I think.

**Review by computate — CHANGES_REQUESTED** [review]
We should squish the commits, maybe combine Zookeeper and Solr as one commit since the Zookeeper PR was never merged.

File: ai-telemetry/base/solr/secret.yaml (hunk 1 above)
Comment (RH-csaggin) [reply ai-telemetry/base/solr/secret.yaml]: - https://github.com/OCP-on-NERC/nerc-ocp-config/pull/547 for Zookeeper was actually merged right? But we can not see it running on the cluster and nether on gitops.
- Yes, I did already implement the externalsecret but I had the secret leftover 😅

**Comment by computate** [comment]
A couple more notes @RH-csaggin , you should create a remote for the upstream repo so that you can pull the latest from the main branch, and fix the conflicts causing the weird duplicate zookeeper resources: 
```bash
git remote add upstream git@github.com:OCP-on-NERC/nerc-ocp-config.git
git pull upstream main
```

**Comment by computate** [comment]
Also rebase and squish some commits up into 1 commit. 
```bash
git rebase -i 4a2f409645cf657a26aa59f3a66aa331f12f166f
```

File: ai-telemetry/base/externalsecrets/solr-externalsecret.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+spec:
+  refreshInterval: 15s
+  secretStoreRef:
+    name: nerc-cluster-secrets
+    kind: ClusterSecretStore
+  target:
+    name: solr-secret
+  dataFrom:
+    - extract:
+        key: $ENV/$CLUSTER/solr-password
```
Comment (computate) [inline ai-telemetry/base/externalsecrets/solr-externalsecret.yaml]: Let's change this to be `key: $ENV/$CLUSTER/ai-telemetry/solr`

File: ai-telemetry/overlays/externalsecrets/solr-externalsecret.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: solr-secret
+  namespace: solr
+spec:
+  dataFrom:
+    - extract:
+        key: nerc/nerc-ocp-obs/solr-secret
```
Comment (computate) [inline ai-telemetry/overlays/externalsecrets/solr-externalsecret.yaml]: Let's change this to be `key: nerc/nerc-ocp-obs/ai-telemetry/solr`.

File: cluster-scope/bundles/solr/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+commonLabels:
+  nerc.mghpcc.org/bundle: solr
+resources:
+- ../../base/core/namespaces/solr
```
Comment (computate) [inline cluster-scope/bundles/solr/kustomization.yaml:6]: We need to reference this bundle in `cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml`

**Review by computate — CHANGES_REQUESTED** [review]

File: ai-telemetry/overlays/externalsecrets/solr-externalsecret.yaml (hunk 3 above)
Comment (computate) [reply ai-telemetry/overlays/externalsecrets/solr-externalsecret.yaml]: I created a vault secret that looks like this with a new password for Solr. 
image

Comment (RH-csaggin) [reply ai-telemetry/overlays/externalsecrets/solr-externalsecret.yaml]: Greate! thanks Chris!

**Comment by RH-csaggin** [comment]
@computate Added modifications as requested :-)

**Review by schwesig — APPROVED** [review]

**Review by computate — APPROVED** [review]

**Pull request merged**